llvm/include/llvm/ADT/APInt.h

```diff
@@ -2188,8 +2188,13 @@ inline const APInt &umax(const APInt &A, const APInt &B) {
   return A.ugt(B) ? A : B;
 }
 
+/// Determine the absolute difference of two APInts considered to be signed.
+inline const APInt abds(const APInt &A, const APInt &B) {
+  return A.sge(B) ? (A - B) : (B - A);
```

Comment:
Don't need the parens (but I'm not insisting you remove them if you prefer them).

Reply:
Thanks, I always tend towards being overly cautious within ternary